miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/api/ApiClient.kt

```diff
@@ -100,14 +105,32 @@ internal class ApiClient @VisibleForTesting constructor(
 internal class RetrofitRequestExecutor(
     private val retrofit: Retrofit
 ) {
-
-    private inline fun <reified T : ErrorResponse> createErrorConvertor(retrofit: Retrofit) =
+    private inline fun <reified T : ErrorResponse> createErrorConverter(retrofit: Retrofit) =
         retrofit.responseBodyConverter<T>(T::class.java, arrayOfNulls<Annotation>(0))
 
-    @Suppress("TooGenericExceptionCaught", "ThrowsCount")
-    suspend fun <T> executeRequest(call: Call<T>): T = try {
+    @Suppress(
+        "TooGenericExceptionCaught", "ThrowsCount", "MagicNumber",
+        "LongMethod", "NestedBlockDepth", "ComplexMethod"
+    )
+    suspend fun <T> executeRequest(call: Call<T>, _retryCount: Int = 0): T = try {
         val response = call.execute()
 
+        // retry network request when there is 500 error code from the server
+        var retryCount = _retryCount
+        if (response.code() >= 500) {
+            if (retryCount++ < TOTAL_RETRIES) {
```

Comment:
You can put only single line of `if` here and remove `NestedBlockDepth`, and perhaps even`ComplexMethod`.